portality/api/v1/crud/articles.py

```diff
@@ -1,5 +1,111 @@
 from portality.api.v1.crud.common import CrudApi
-
+from portality.api.v1 import Api400Error, Api401Error, Api403Error, Api404Error
+from portality.api.v1.data_objects import IncomingArticleDO, OutgoingArticleDO
+from portality.lib import dataobj
+from portality import models
+from portality.article import XWalk
 
 class ArticlesCrudApi(CrudApi):
-    pass
\ No newline at end of file
+
+    @classmethod
+    def create(cls, data, account):
+        # as long as authentication (in the layer above) has been successful, and the account exists, then
+        # we are good to proceed
+        if account is None:
+            raise Api401Error()
+
+        # first thing to do is a structural validation, by instantiating the data object
+        try:
+            ia = IncomingArticleDO(data)
+        except dataobj.DataStructureException as e:
+            raise Api400Error(e.message)
+
+        # if that works, convert it to an Article object
+        am = ia.to_article_model()
+
+        # Check we are allowed to create an article for this journal
+        if not XWalk.is_legitimate_owner(am, account.id):
+            raise Api404Error()                 # not found for this account
+
+        # if the caller set the id, created_date, or last_updated, then we discard the data and apply our
+        # own values (note that last_updated will get overwritten anyway)
+        am.set_id()
+        am.set_created()
```

Comment:
Do we also need to overwrite the "journal" section of the article bibjson? Except start_page, end_page, volume and number (issue number), the other stuff (like title, license, country) will need to be copied in from the journal I think. Even if we let people specify them, we should ignore them. It's not possible (or shouldn't be) to upload an article without a journal being already present, so we can always perform this overwrite. It is also only happening on create and update, on the way into the system.